README.md
# CryptoTracker v0.1.0 GPLv3

Simple HTML5/Javascript cryptoportfolio tracker, pulls data from Poloniex.com public API and charts it.

![Screenshot](screenshot.png)

# Installation and Running

```
$ git clone git@github.com:wolfmankurd/CryptoTracker.git
$ cd CryptoTracker
$ npm install request
$ node CryptoTracker.js
```

# Dependencies

* NodeJS and request module.
* [JQuery](http://jquery.com/) using 3.2.1 hosted via Googleapis.com
* [DojiChart](http://dojichart.com/)



CHANGELOG.md
# Changelog

* 2018-01-14
	- Change to Node
* 2018-01-12
	- Added crosshair
	- Added changelog
	- Added scroll button
	- First working version


poloData.js
var poloData = [{"volume": 1.81630424, "quoteVolume": 67.15015667, "high": 0.02706937, "low": 0.02692003, "time": "2018-01-12T06:25:00Z", "date": 1515738300, "close": 0.02706936, "weightedAverage": 0.02704839, "open": 0.02692003}, {"volume": 3.29849152, "quoteVolume": 122.0656535, "high": 0.02708895, "low": 0.02696326, "time": "2018-01-12T06:30:00Z", "date": 1515738600, "close": 0.02698, "weightedAverage": 0.02702227, "open": 0.02701}, {"volume": 0.36695319, "quoteVolume": 13.63981528, "high": 0.02698, "low": 0.02685, "time": "2018-01-12T06:35:00Z", "date": 1515738900, "close": 0.02690345, "weightedAverage": 0.02690309, "open": 0.02698}, {"volume": 0.66965625, "quoteVolume": 24.83929261, "high": 0.02697987, "low": 0.02690348, "time": "2018-01-12T06:40:00Z", "date": 1515739200, "close": 0.02695997, "weightedAverage": 0.02695955, "open": 0.02692296}, {"volume": 2.49721597, "quoteVolume": 92.72120889, "high": 0.02696, "low": 0.02684, "time": "2018-01-12T06:45:00Z", "date": 1515739500, "close": 0.02684001, "weightedAverage": 0.02693252, "open": 0.02692396}, {"volume": 0.72885796, "quoteVolume": 27.11407319, "high": 0.02695497, "low": 0.02684, "time": "2018-01-12T06:50:00Z", "date": 1515739800, "close": 0.02684025, "weightedAverage": 0.02688116, "open": 0.02684}, {"volume": 1.57222603, "quoteVolume": 58.58903452, "high": 0.026913, "low": 0.026801, "time": "2018-01-12T06:55:00Z", "date": 1515740100, "close": 0.026801, "weightedAverage": 0.02683481, "open": 0.026903}, {"volume": 1.84535612, "quoteVolume": 68.6209045, "high": 0.02695993, "low": 0.0268, "time": "2018-01-12T07:00:00Z", "date": 1515740400, "close": 0.02690796, "weightedAverage": 0.02689204, "open": 0.02681776}, {"volume": 2.98969321, "quoteVolume": 110.83558081, "high": 0.02701752, "low": 0.026885, "time": "2018-01-12T07:05:00Z", "date": 1515740700, "close": 0.02689, "weightedAverage": 0.02697412, "open": 0.02695978}, {"volume": 0.26729303, "quoteVolume": 9.93449661, "high": 0.02701639, "low": 0.02689001, "time": "2018-01-12T07:10:00Z", "date": 1515741000, "close": 0.02695001, "weightedAverage": 0.02690554, "open": 0.02689001}, {"volume": 0.96295301, "quoteVolume": 35.72914203, "high": 0.02701368, "low": 0.02695, "time": "2018-01-12T07:15:00Z", "date": 1515741300, "close": 0.02695268, "weightedAverage": 0.02695147, "open": 0.02695}, {"volume": 0.23169066, "quoteVolume": 8.5931159, "high": 0.02699699, "low": 0.02695768, "time": "2018-01-12T07:20:00Z", "date": 1515741600, "close": 0.02699696, "weightedAverage": 0.02696235, "open": 0.02695768}, {"volume": 0.68716302, "quoteVolume": 25.48398726, "high": 0.02699699, "low": 0.02695005, "time": "2018-01-12T07:25:00Z", "date": 1515741900, "close": 0.02695005, "weightedAverage": 0.0269645, "open": 0.02699693}, {"volume": 1.20814834, "quoteVolume": 44.8375618, "high": 0.02697625, "low": 0.026885, "time": "2018-01-12T07:30:00Z", "date": 1515742200, "close": 0.02688597, "weightedAverage": 0.026945, "open": 0.02695005}, {"volume": 2.26131258, "quoteVolume": 84.12472936, "high": 0.02694, "low": 0.0268, "time": "2018-01-12T07:35:00Z", "date": 1515742500, "close": 0.02680047, "weightedAverage": 0.02688047, "open": 0.02688597}, {"volume": 6.75155945, "quoteVolume": 249.9333221, "high": 0.0270632, "low": 0.02683176, "time": "2018-01-12T07:40:00Z", "date": 1515742800, "close": 0.02703496, "weightedAverage": 0.02701344, "open": 0.02694}, {"volume": 4.68335489, "quoteVolume": 173.41068085, "high": 0.02709328, "low": 0.02683131, "time": "2018-01-12T07:45:00Z", "date": 1515743100, "close": 0.02695794, "weightedAverage": 0.0270073, "open": 0.02703496}, {"volume": 1.18235852, "quoteVolume": 43.97532709, "high": 0.02695801, "low": 0.02680003, "time": "2018-01-12T07:50:00Z", "date": 1515743400, "close": 0.02681508, "weightedAverage": 0.02688686, "open": 0.02691041}, {"volume": 1.54571031, "quoteVolume": 57.49926153, "high": 0.0269, "low": 0.02682004, "time": "2018-01-12T07:55:00Z", "date": 1515743700, "close": 0.02686013, "weightedAverage": 0.02688226, "open": 0.02689488}, {"volume": 9.05067883, "quoteVolume": 336.19207558, "high": 0.02709328, "low": 0.02680703, "time": "2018-01-12T08:00:00Z", "date": 1515744000, "close": 0.02691003, "weightedAverage": 0.02692115, "open": 0.02686013}, {"volume": 5.07729205, "quoteVolume": 187.03598432, "high": 0.02722, "low": 0.026915, "time": "2018-01-12T08:05:00Z", "date": 1515744300, "close": 0.02707715, "weightedAverage": 0.02714607, "open": 0.02691591}, {"volume": 2.78746165, "quoteVolume": 103.12223531, "high": 0.02710797, "low": 0.02695015, "time": "2018-01-12T08:10:00Z", "date": 1515744600, "close": 0.02695015, "weightedAverage": 0.02703065, "open": 0.027093}, {"volume": 0.19667222, "quoteVolume": 7.29624113, "high": 0.02699988, "low": 0.0269, "time": "2018-01-12T08:15:00Z", "date": 1515744900, "close": 0.026931, "weightedAverage": 0.02695528, "open": 0.02695}, {"volume": 0.51472659, "quoteVolume": 19.09390064, "high": 0.026995, "low": 0.02691, "time": "2018-01-12T08:20:00Z", "date": 1515745200, "close": 0.02691, "weightedAverage": 0.02695764, "open": 0.026931}, {"volume": 0.72960876, "quoteVolume": 27.09511507, "high": 0.02695, "low": 0.026893, "time": "2018-01-12T08:25:00Z", "date": 1515745500, "close": 0.026893, "weightedAverage": 0.02692768, "open": 0.02694997}, {"volume": 0.91144289, "quoteVolume": 33.88745666, "high": 0.02697436, "low": 0.02680703, "time": "2018-01-12T08:30:00Z", "date": 1515745800, "close": 0.02680703, "weightedAverage": 0.02689617, "open": 0.02695997}, {"volume": 0.68770243, "quoteVolume": 25.61469091, "high": 0.02692722, "low": 0.02680703, "time": "2018-01-12T08:35:00Z", "date": 1515746100, "close": 0.02683204, "weightedAverage": 0.02684796, "open": 0.026809}, {"volume": 0.34748203, "quoteVolume": 12.93094811, "high": 0.02692497, "low": 0.02684202, "time": "2018-01-12T08:40:00Z", "date": 1515746400, "close": 0.02688597, "weightedAverage": 0.02687212, "open": 0.02684202}, {"volume": 1.55156496, "quoteVolume": 57.86517554, "high": 0.02688597, "low": 0.0267, "time": "2018-01-12T08:45:00Z", "date": 1515746700, "close": 0.0267, "weightedAverage": 0.02681344, "open": 0.02688597}, {"volume": 0.52841033, "quoteVolume": 19.74006722, "high": 0.026825, "low": 0.026665, "time": "2018-01-12T08:50:00Z", "date": 1515747000, "close": 0.026779, "weightedAverage": 0.02676841, "open": 0.0267891}, {"volume": 1.17068561, "quoteVolume": 43.73475515, "high": 0.02681999, "low": 0.02669501, "time": "2018-01-12T08:55:00Z", "date": 1515747300, "close": 0.02679997, "weightedAverage": 0.02676785, "open": 0.02681493}, {"volume": 1.16299459, "quoteVolume": 43.56115968, "high": 0.02678942, "low": 0.02666002, "time": "2018-01-12T09:00:00Z", "date": 1515747600, "close": 0.02673, "weightedAverage": 0.02669797, "open": 0.02678942}, {"volume": 1.29401287, "quoteVolume": 48.43723086, "high": 0.02682297, "low": 0.02666245, "time": "2018-01-12T09:05:00Z", "date": 1515747900, "close": 0.02666511, "weightedAverage": 0.02671525, "open": 0.02666245}, {"volume": 2.14583034, "quoteVolume": 80.29750316, "high": 0.02682999, "low": 0.02666519, "time": "2018-01-12T09:10:00Z", "date": 1515748200, "close": 0.02676703, "weightedAverage": 0.0267235, "open": 0.02666519}, {"volume": 1.55562514, "quoteVolume": 58.04800249, "high": 0.02683997, "low": 0.02673303, "time": "2018-01-12T09:15:00Z", "date": 1515748500, "close": 0.02677424, "weightedAverage": 0.02679894, "open": 0.02678209}, {"volume": 1.10323705, "quoteVolume": 41.23815748, "high": 0.02683426, "low": 0.02670596, "time": "2018-01-12T09:20:00Z", "date": 1515748800, "close": 0.02671038, "weightedAverage": 0.02675282, "open": 0.02683426}, {"volume": 1.87645783, "quoteVolume": 70.32564814, "high": 0.02679796, "low": 0.02667603, "time": "2018-01-12T09:25:00Z", "date": 1515749100, "close": 0.02677796, "weightedAverage": 0.02668241, "open": 0.02671041}, {"volume": 5.11948585, "quoteVolume": 190.95484039, "high": 0.02699238, "low": 0.02667664, "time": "2018-01-12T09:30:00Z", "date": 1515749400, "close": 0.02684797, "weightedAverage": 0.02680992, "open": 0.02667664}, {"volume": 2.19303591, "quoteVolume": 81.84223765, "high": 0.02684796, "low": 0.02673403, "time": "2018-01-12T09:35:00Z", "date": 1515749700, "close": 0.02676, "weightedAverage": 0.02679589, "open": 0.02684294}, {"volume": 1.00429883, "quoteVolume": 37.57356158, "high": 0.02683291, "low": 0.02670801, "time": "2018-01-12T09:40:00Z", "date": 1515750000, "close": 0.02670803, "weightedAverage": 0.02672886, "open": 0.02676}, {"volume": 0.90317278, "quoteVolume": 33.80224101, "high": 0.02684296, "low": 0.026666, "time": "2018-01-12T09:45:00Z", "date": 1515750300, "close": 0.026808, "weightedAverage": 0.02671931, "open": 0.02677293}, {"volume": 1.79456555, "quoteVolume": 67.06351596, "high": 0.0268, "low": 0.02666603, "time": "2018-01-12T09:50:00Z", "date": 1515750600, "close": 0.02676006, "weightedAverage": 0.02675919, "open": 0.02666603}, {"volume": 2.11429652, "quoteVolume": 78.76185295, "high": 0.02690899, "low": 0.02672184, "time": "2018-01-12T09:55:00Z", "date": 1515750900, "close": 0.02672613, "weightedAverage": 0.02684416, "open": 0.02676022}, {"volume": 2.06331808, "quoteVolume": 76.99306866, "high": 0.02691197, "low": 0.02672649, "time": "2018-01-12T10:00:00Z", "date": 1515751200, "close": 0.02681501, "weightedAverage": 0.02679875, "open": 0.02672649}, {"volume": 0.69493406, "quoteVolume": 25.90327353, "high": 0.02687479, "low": 0.02677906, "time": "2018-01-12T10:05:00Z", "date": 1515751500, "close": 0.02677906, "weightedAverage": 0.02682804, "open": 0.02683503}, {"volume": 0.90646363, "quoteVolume": 33.78495262, "high": 0.02690744, "low": 0.02676808, "time": "2018-01-12T10:10:00Z", "date": 1515751800, "close": 0.02687658, "weightedAverage": 0.02683039, "open": 0.02684994}, {"volume": 1.08009398, "quoteVolume": 40.39323491, "high": 0.02687652, "low": 0.02665455, "time": "2018-01-12T10:15:00Z", "date": 1515752100, "close": 0.02673794, "weightedAverage": 0.02673947, "open": 0.02687652}, {"volume": 3.91937732, "quoteVolume": 146.82167508, "high": 0.0267884, "low": 0.02663, "time": "2018-01-12T10:20:00Z", "date": 1515752400, "close": 0.02663, "weightedAverage": 0.02669481, "open": 0.02665456}, {"volume": 0.49964035, "quoteVolume": 18.73511824, "high": 0.02677296, "low": 0.02663031, "time": "2018-01-12T10:25:00Z", "date": 1515752700, "close": 0.0267383, "weightedAverage": 0.02666865, "open": 0.02663031}, {"volume": 2.62085859, "quoteVolume": 97.65373246, "high": 0.02694193, "low": 0.02668164, "time": "2018-01-12T10:30:00Z", "date": 1515753000, "close": 0.02675501, "weightedAverage": 0.02683828, "open": 0.0267383}, {"volume": 1.91360549, "quoteVolume": 71.18414697, "high": 0.026958, "low": 0.0267777, "time": "2018-01-12T10:35:00Z", "date": 1515753300, "close": 0.02677785, "weightedAverage": 0.02688246, "open": 0.02694299}, {"volume": 0.98293232, "quoteVolume": 36.54142442, "high": 0.02696697, "low": 0.026798, "time": "2018-01-12T10:40:00Z", "date": 1515753600, "close": 0.026937, "weightedAverage": 0.02689912, "open": 0.026798}, {"volume": 1.2690122, "quoteVolume": 47.28879024, "high": 0.02689309, "low": 0.0267, "time": "2018-01-12T10:45:00Z", "date": 1515753900, "close": 0.02670001, "weightedAverage": 0.02683537, "open": 0.0268795}, {"volume": 1.80936138, "quoteVolume": 67.66965543, "high": 0.02684599, "low": 0.02668, "time": "2018-01-12T10:50:00Z", "date": 1515754200, "close": 0.02668012, "weightedAverage": 0.02673814, "open": 0.026705}, {"volume": 1.06756996, "quoteVolume": 39.8896173, "high": 0.026821, "low": 0.02668018, "time": "2018-01-12T10:55:00Z", "date": 1515754500, "close": 0.02680591, "weightedAverage": 0.0267631, "open": 0.02674599}, {"volume": 0.22754652, "quoteVolume": 8.52175826, "high": 0.026852, "low": 0.02669695, "time": "2018-01-12T11:00:00Z", "date": 1515754800, "close": 0.02670129, "weightedAverage": 0.02670182, "open": 0.02669695}, {"volume": 7.08299254, "quoteVolume": 266.21933904, "high": 0.02679699, "low": 0.02650508, "time": "2018-01-12T11:05:00Z", "date": 1515755100, "close": 0.02650508, "weightedAverage": 0.02660585, "open": 0.02670148}, {"volume": 1.94511452, "quoteVolume": 73.38779552, "high": 0.02659395, "low": 0.02647413, "time": "2018-01-12T11:10:00Z", "date": 1515755400, "close": 0.02647413, "weightedAverage": 0.0265046, "open": 0.02650509}, {"volume": 0.75081372, "quoteVolume": 28.28905825, "high": 0.02656375, "low": 0.02646263, "time": "2018-01-12T11:15:00Z", "date": 1515755700, "close": 0.02655897, "weightedAverage": 0.02654078, "open": 0.02651896}, {"volume": 2.06707026, "quoteVolume": 77.66186049, "high": 0.02674011, "low": 0.026511, "time": "2018-01-12T11:20:00Z", "date": 1515756000, "close": 0.02674011, "weightedAverage": 0.02661628, "open": 0.02660982}, {"volume": 3.46427063, "quoteVolume": 129.48779754, "high": 0.02681599, "low": 0.02665095, "time": "2018-01-12T11:25:00Z", "date": 1515756300, "close": 0.02680597, "weightedAverage": 0.02675364, "open": 0.026735}, {"volume": 3.13588912, "quoteVolume": 116.97915844, "high": 0.02684102, "low": 0.02671938, "time": "2018-01-12T11:30:00Z", "date": 1515756600, "close": 0.026784, "weightedAverage": 0.02680724, "open": 0.026751}, {"volume": 2.58486829, "quoteVolume": 96.57792111, "high": 0.02684102, "low": 0.02665789, "time": "2018-01-12T11:35:00Z", "date": 1515756900, "close": 0.026836, "weightedAverage": 0.02676458, "open": 0.026789}, {"volume": 9.29619462, "quoteVolume": 346.81408787, "high": 0.02689146, "low": 0.02662333, "time": "2018-01-12T11:40:00Z", "date": 1515757200, "close": 0.026815, "weightedAverage": 0.02680454, "open": 0.0268}, {"volume": 0.53388639, "quoteVolume": 19.9029752, "high": 0.02688404, "low": 0.02675715, "time": "2018-01-12T11:45:00Z", "date": 1515757500, "close": 0.02681706, "weightedAverage": 0.02682445, "open": 0.02681019}, {"volume": 0.72300429, "quoteVolume": 26.86166415, "high": 0.02694191, "low": 0.02676703, "time": "2018-01-12T11:50:00Z", "date": 1515757800, "close": 0.02694191, "weightedAverage": 0.02691584, "open": 0.02682703}, {"volume": 0.22486397, "quoteVolume": 8.36475946, "high": 0.02694188, "low": 0.02681703, "time": "2018-01-12T11:55:00Z", "date": 1515758100, "close": 0.02685701, "weightedAverage": 0.02688229, "open": 0.02687069}, {"volume": 7.30208243, "quoteVolume": 272.12023832, "high": 0.02698238, "low": 0.02675708, "time": "2018-01-12T12:00:00Z", "date": 1515758400, "close": 0.02682988, "weightedAverage": 0.02683402, "open": 0.026912}, {"volume": 1.04873504, "quoteVolume": 39.04706401, "high": 0.0269824, "low": 0.02683007, "time": "2018-01-12T12:05:00Z", "date": 1515758700, "close": 0.02683346, "weightedAverage": 0.02685823, "open": 0.02683007}, {"volume": 1.39736884, "quoteVolume": 52.08198273, "high": 0.02693705, "low": 0.02677487, "time": "2018-01-12T12:10:00Z", "date": 1515759000, "close": 0.0268, "weightedAverage": 0.02683017, "open": 0.02688003}, {"volume": 1.55395113, "quoteVolume": 57.77837409, "high": 0.02699633, "low": 0.02680006, "time": "2018-01-12T12:15:00Z", "date": 1515759300, "close": 0.02699238, "weightedAverage": 0.02689503, "open": 0.02690655}];


web/CryptoTracker.js
dojichart = new DojiChart.core.Chart(document.getElementById('my-chart'), {
    width: 700,
    fieldMap: {
        time: 'time',
        open: 'open',
        high: 'high',
        low: 'low',
        close: 'close',
        volume: 'volume'
    },
    crosshair: true
});
// Price
var price_chart_panel = new DojiChart.panel.TimeValuePanel({
  primaryLayer: new DojiChart.layer.CandleLayer({}),
  height: 200,
  grid: true
});

dojichart.addComponent("price", price_chart_panel);

// SMA 200
var sma200_layer = new DojiChart.layer.indicator.SimpleMovingAverageLayer({
  period: 200,
  color: "blue"
});
price_chart_panel.addLayer(sma200_layer);

// SMA 50
var sma50_layer = new DojiChart.layer.indicator.SimpleMovingAverageLayer({
  period: 50,
  color: "orange"
});
price_chart_panel.addLayer(sma50_layer);

// EMA 10
var ema10_layer = new DojiChart.layer.indicator.ExponentialMovingAverageLayer({
  period: 10,
  color: "fuchsia"
});
price_chart_panel.addLayer(ema10_layer);

// Bollinger Bands
var bb_layer = new DojiChart.layer.indicator.BollingerBandsLayer({
  midColor: "rgba(0, 128, 0, 0.4)",
  bandColor: "rgba(0, 0, 0, 0.3)" 
});
price_chart_panel.addLayer(bb_layer);

// Volume profile
var volume_profile_layer = new DojiChart.layer.indicator.VolumeProfileLayer({
});
price_chart_panel.addLayer(volume_profile_layer);

// Time labels (at top of chart)
var time_labels_panel = new DojiChart.panel.TimeLabelsPanel();
dojichart.addComponent("timelabels", time_labels_panel);

// Volume panel
var volume_chart_panel = new DojiChart.panel.TimeValuePanel({
  height: 100,
  primaryAtBack: true,
  primaryLayer: new DojiChart.layer.indicator.VolumeLayer({
    layerIndex: 1,
    barColor: "#3377FF",
    barWidth: 5
  })
});

dojichart.addComponent("volume", volume_chart_panel, true);

// Volume EMA 65
var ema65_layer = new DojiChart.layer.indicator.ExponentialMovingAverageLayer({
  layerIndex: 2,
  input: "volume",
  period: 65,
  color: "black"
});
volume_chart_panel.addLayer(ema65_layer);

// Stochastic panel
var stochastic_chart_panel = new DojiChart.panel.TimeValuePanel({
  height: 68,
  grid: {
    value: {
      lines: [20, 50, 80]
    }
  },
  primaryLayer: new DojiChart.layer.indicator.StochasticLayer({
  })
});

dojichart.addComponent("stochastic", stochastic_chart_panel, true);

// RSI panel
var rsi_chart_panel = new DojiChart.panel.TimeValuePanel({
  height: 68,
  grid: {
    value: {
      lines: [30, 70]
    }
  },
  primaryLayer: new DojiChart.layer.indicator.RSILayer({
  })
});

dojichart.addComponent("rsi", rsi_chart_panel, true);

function updateChart(){
    var i;
    for(i = 0; i < poloData.length; i++){
    }
    dojichart.loadData(poloData, "Poloniex.com", 'M5');    
}

updateChart();
setInterval(updateChart, 1*60*1000);

$("#btn-bak").click(function(){
    dojichart.scroll(dojichart.getOffset() - 10, "false");
})
$("#btn-fwd").click(function(){
    dojichart.scroll(dojichart.getOffset() + 10, "false");
})
